code/XFileExporter.cpp

```diff
@@ -514,12 +514,16 @@ void XFileExporter::WriteMesh(aiMesh* mesh)
 
 std::string XFileExporter::toXFileString(aiString &name)
 {
-	std::string str = std::string(name.C_Str());
-	std::replace(str.begin(), str.end(), '<', '_');
-	std::replace(str.begin(), str.end(), '>', '_');
-	std::replace(str.begin(), str.end(), '{', '_');
-	std::replace(str.begin(), str.end(), '}', '_');
-	std::replace(str.begin(), str.end(), '$', '_');
+	std::string pref = "NN_"; // node name prefix to prevent unexpected start of string
```

Comment:
Please doublecheck. If assimp's x importer is able to read it with a leading underscore and there is no other importer who can't then we should allow it.

Otherwise, please add the prefix only if the name begins with an underscore.